invenio_mail/tasks.py

```diff
@@ -27,7 +30,52 @@ def send_email(data):
        `custom serializer <http://docs.celeryproject.org/en/latest/
        userguide/calling.html#serializers>`__
        can be created if attachments are really needed.
+
+       This version adds attachments by putting them into a base64 encoded string.
+       This is not an optimal solution, it might get problematic if they are too
+       large to handle by the messaging queue, so check for a maximum size beforehand.
     """
     msg = Message()
-    msg.__dict__.update(data)
+    msg.subject = data.get("subject")
+    if "html" in data:
+        msg.html = data.get("html")
+
+    msg.body = data.get("body")
+    msg.recipients = data.get("recipients")
+    msg.sender = data.get("sender")
+    if "reply_to" in data:
+        msg.reply_to = data.get("reply_to")
+
+    if "date" in data:
+        msg.date = data.get("date")
+
+    if "cc" in data:
+        msg.cc = data.get("cc")
+
+    if "bcc" in data:
+        msg.cc = data.get("bcc")
+
+    if "charset" in data:
+        msg.cc = data.get("charset")
+
+    if "extra_headers" in data:
+        msg.cc = data.get("extra_headers")
+
+    if "mail_options" in data:
+        msg.cc = data.get("mail_options")
+
+    if "rcpt_options" in data:
+        msg.cc = data.get("rcpt_options")
+
+    if "attachments" in data:
+        for attachment in data.get("attachments"):
+            rawdata = b64decode(attachment.get("base64"))
```

Comment:
What about limiting the max size?

Reply:
I was not sure about this: this code is called after the attachment has been pulled from the message queue, so I felt it made no sense to do the check here.

Reply:
after discussions (and sorry if we did not think about that earlier), maybe a better solution would:
1. create a new func that accept attachments
2. the new method will check the size
3. if valid, it will call an async task


```python
def send_email_attachments(data, attachments):
    """Nice docstring..."""
    for att in attachments:
        ... if att size > config["MAX_ATTACHMENT_SIZE"]:
            fail
    ... update data dict and add attachments ...
    _send_email_attachments.apply_async(data)


@shared_task
def _send_email_attachments
    msg = Message()
    msg.__dict__.update(data)
    msg.attach(content_type=content_type, data=rawdata, disposition=disposition)
    ...
```
WDYT?

Reply:
Sounds reasonable, but I'm not quite sure I completely understand your suggestion. The first function is not called via celery but directly? And by calling `apply_async` a task will be generated? Ok, that would eliminate my objection.

Reply:
Added a check for size limit and a test to see if it works.

Reply:
> Sounds reasonable, but I'm not quite sure I completely understand your suggestion. The first function is not called via celery but directly? And by calling `apply_async` a task will be generated? Ok, that would eliminate my objection.

You implemented it exactly as I was thinking 🥳